src/App.tsx

```diff
@@ -1,14 +1,68 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilteredBy } from './types/Filter';
+import { getTodos } from './api';
+
+const filteredTodos = (todos: Todo[], query: string, filterBy: FilteredBy) => {
+  let readyTodos = [...todos];
+
+  if (query) {
+    const readyQuery = query.toLowerCase().trim();
+
+    readyTodos = readyTodos.filter(todo => todo.title.toLowerCase()
+      .includes(readyQuery));
+  }
+
+  switch (filterBy) {
+    case FilteredBy.ACTIVE:
+      return readyTodos.filter(todo => !todo.completed);
+    case FilteredBy.COMPLETED:
+      return readyTodos.filter(todo => todo.completed);
+    default:
+      return readyTodos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.ALL);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const visibleTodos = filteredTodos(todos, query, filterBy);
+
```

Comment:
My assumption is name util `getVisibleTodos`

Reply:
isn't fixed. Functions name should start with a verb